server/api/dataset.py

```diff
@@ -89,8 +96,7 @@ def getDataset(self, dataset, params):
 
     @describeRoute(
         Description('Create a lesion image dataset.')
-        .param('uploadFolderId', 'The ID of the folder that contains images '
-               'and metadata.')
+        .param('uploadFolderId', 'The ID of the folder that contains images.')
```

Comment:
Change this into `uploadFileId`, and have it point to a single Girder file, which should greatly simplify the validation here (and in the `Dataset.ingestDataset` method).